File: linear_search/linear_search.cpp
#include<iostream>

using namespace std;

// This function performs linear search based on the given input
/* Parameters:
   arr[] -> array of type int
   arr_length -> the length of the array
   search_item -> the number to be searched
*/
// This function returns an integer value
// -1 if search item not found
// index of the array where the search item is found
int linear_search(int arr[], int arr_length, int search_item){

  for(int i = 0; i <= arr_length; i++){
    if(arr[i] == search_item)
      return i;
  }
  return -1;

}
// Driver function
int main(){
  
  // declare an array
  int arr[] = {19,39,59,68,32,35,12,34};
  // find array length
  int arr_len = sizeof(arr)/sizeof(arr[0]);
  // get a number to search
  int search_element = 12;

  int result = linear_search(arr, arr_len, search_element);

  if(result == -1)
    cout << search_element << " is not found in the array"; 
  else
    cout << search_element <<" is found at " <<  result + 1 << "th position in the array" ; 

  return 0;

}

File: linear_search/linear_search.js
// This program uses linear search algorithm to search an element in an arrau

function linear_search(arr, search_element) {
  /*
    @param:   arr             "An array"                                   int
    @param:   search_element  "Element to be searched with"                int
    @return: -1               "If the element is not found"                int
    @return:  i               "Index of array where the element is found"  int
    */

  let arr_len = arr.length;

  for (let i = 0; i < arr_len; i++) {
    if (arr[i] === search_element) {
      return i;
    }
  }
  return -1;
}

// Driver function
const arr = [19, 39, 59, 68, 32, 35, 12, 34];
const search_item = 12;
let result = linear_search(arr, search_item);

if (result === -1) {
  console.log(`${search_item} is not found`);
} else {
  console.log(`${search_item} is found at ${result + 1}th positon`);
}
